### PR TITLE
steam container

### PR DESCRIPTION
**Pull request by Eibironke**
Added:

Tests 
Dockerfile
Folder to contain steam related scripts

for steam_extract.py

### CODE REVIEW

File: pipeline/extract/steam/conftest.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import pytest
```
Comment (annalisev) [inline pipeline/extract/steam/conftest.py:1]: Docstring at the start?

File: pipeline/extract/steam/conftest.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+from bs4 import BeautifulSoup
+
+
+@pytest.fixture
+def steam_rating_soup():
+    return BeautifulSoup(""" <div class="review_ctn"> 
+                         <label for="review_type_positive">Positive&nbsp;<span class="user_reviews_count">(105)</span></label><br> 
+                         <label for="review_type_negative">Negative&nbsp;<span class="user_reviews_count">(103)</span></label>
+                         </div>""", features='html.parser')
```
Comment (annalisev) [inline pipeline/extract/steam/conftest.py:11]: Nice use of fixtures

File: pipeline/extract/steam/test_extract_steam.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
+"""Test file for steam_extract.py"""
```
Comment (annalisev) [inline pipeline/extract/steam/test_extract_steam.py:1]:
```suggestion
"""Test file for steam_extract.py."""
```

File: pipeline/extract/steam/test_extract_steam.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
+"""Test file for steam_extract.py."""
+
+from steam_extract import get_rating, get_platform_ids, get_tags, get_developer, get_publisher, get_name_price_date
```
Comment (annalisev) [inline pipeline/extract/steam/test_extract_steam.py:3]:
```suggestion
from steam_extract import (get_rating, get_platform_ids,
get_tags, get_developer, get_publisher, get_name_price_date)
```

Comment (annalisev) [reply pipeline/extract/steam/test_extract_steam.py:3]: To stop the line being so long?

File: pipeline/extract/steam/test_extract_steam.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+
+# ratings tests
+def test_get_ratings(steam_rating_soup):
+    '''Tests the get_ratings function with standard inputs.'''
+    assert get_rating(steam_rating_soup) == 50.48
+
+
+def test_get_ratings_no_reviews(steam_no_rating_soup):
+    '''Tests the get_ratings function with no input/reviews'''
```
Comment (annalisev) [inline pipeline/extract/steam/test_extract_steam.py:13]:
```suggestion
    '''Tests the get_ratings function with no input/reviews.'''
```

File: pipeline/extract/steam/test_extract_steam.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+
+def test_get_tags(steam_tags_good):
+    """Test the get_tags function with all good inputs"""
+    assert get_tags(steam_tags_good) == ["Horror", "Atmospheric"]
+
+
+def test_get_tags_no_tags(steam_tags_none):
+    """Test the get_tags function with no tags"""
+    assert get_tags(steam_tags_none) == []
```
Comment (annalisev) [inline pipeline/extract/steam/test_extract_steam.py:50]:
```suggestion
    """Test the get_platform function with all good inputs."""
    list_p = ["Windows", "macOS", "Linux"]
    assert get_platform_ids(list_p) == [1, 2, 3]


def test_get_platform_id_bad_input():
    """Test the get_platform function with all good inputs."""
    list_b = ["Widow", "bigmac", "Lemur"]
    assert get_platform_ids(list_b) == []

# get_tags tests


def test_get_tags(steam_tags_good):
    """Test the get_tags function with all good inputs."""
    assert get_tags(steam_tags_good) == ["Horror", "Atmospheric"]


def test_get_tags_no_tags(steam_tags_none):
    """Test the get_tags function with no tags."""
    assert get_tags(steam_tags_none) == []
```

File: pipeline/extract/steam/test_extract_steam.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+def test_publisher_missing(steam_developer_none):
+    """Test the get_developer function with a no input"""
+    assert get_developer(steam_developer_none) is None
+
+
+# name_price tests
+
+def test_get_name_price_date(search_result):
+    """Test the get_name_price_date function is working"""
+    assert get_name_price_date(search_result) == [
```
Comment (annalisev) [inline pipeline/extract/steam/test_extract_steam.py:82]:
```suggestion
def test_developer(steam_developer_good):
    """Test the get_developer function with a good inputs."""
    assert get_developer(steam_developer_good) == "SCLITIFY IT "


def test_developer_missing(steam_developer_none):
    """Test the get_developer function with a good inputs."""
    assert get_developer(steam_developer_none) is None


# get_publisher tests


def test_publisher(steam_publisher_good):
    """Test the get_developer function with a good inputs."""
    assert get_publisher(steam_publisher_good) == "Inoa Forever"


def test_publisher_missing(steam_developer_none):
    """Test the get_developer function with a no input."""
    assert get_developer(steam_developer_none) is None


# name_price tests

def test_get_name_price_date(search_result):
    """Test the get_name_price_date function is working."""
    assert get_name_price_date(search_result) == [
```

**Review by annalisev — CHANGES_REQUESTED** [review]
A few suggestions but otherwise looks good!

File: pipeline/extract/steam/conftest.py (hunk 1 above)
Comment (Eibironke) [reply pipeline/extract/steam/conftest.py:1]: Good shout

File: pipeline/extract/steam/conftest.py [hunk 8]
```diff
@@ -0,0 +1,95 @@
+"""conftest file for test_extract_steam"""
```
Comment (annalisev) [inline pipeline/extract/steam/conftest.py:1]:
```suggestion
"""Conftest file for test_extract_steam."""
```

**Review by annalisev — APPROVED** [review]
Looks good!

**Pull request merged**